frame/support/src/dispatch.rs

```diff
@@ -1325,13 +1331,27 @@ macro_rules! decl_module {
 				$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!("on_runtime_upgrade"));
 				let result: $return = (|| { $( $impl )* })();
 
-				$crate::crate_to_pallet_version!()
+				let new_storage_version = $crate::crate_to_pallet_version!();
+				new_storage_version
 					.put_into_storage::<<$trait_instance as $system::Config>::PalletInfo, Self>();
 
 				let additional_write = <
 					<$trait_instance as $system::Config>::DbWeight as $crate::traits::Get<_>
 				>::get().writes(1);
 
+				let pallet_name = <<
+					$trait_instance
+					as
+					$system::Config
+				>::PalletInfo as $crate::traits::PalletInfo>::name::<Self>().expect("pallet will have name in the runtime.");
+
+				$crate::debug::info!(
+					target: $crate::LOG_TARGET,
```

Comment:
Shouldn't this be under the target of the pallet?

Reply:
Then we would need to assume that each pallet is exposing a `LOG_TARGET` or some similar identifier. I actually propsed it earlier in this PR, but it didn't gain any attention :(  https://github.com/paritytech/substrate/pull/8128#issuecomment-779293607

Reply:
Ahh the link there is now broken, in this commit 
 if you look for TODO you will find my note that: https://github.com/paritytech/substrate/pull/8123/commits/87baef8be0d857e162f5202d4cd890333ca66f55#diff-6faadaafc4a2f7556e8ba9cf9bd8544e5c039e97441e1d470a88c296d39819edR1378-R1383
 
 ```
				// TODO: idea: we can enforce each pallet to define a LOG_TARGET const, for which
				// the default is crate_name!(). We prefix it automatically with `runtime::foo` to
				// have unified logging. Lastly, we  could provide an additional macro like the one
				// I have put in staking for easier logging. Then, we could use the real crate's
				// LOG_TARGET here, not the one from frame-support.
```

Reply:
I think we should provide some macro `log_target!(Self)` that is concatenating `runtime` with the name of `Self` in the runtime (by using `PalletInfo`)

Reply:
👍 woudl also be good